packages/core/src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -129,12 +129,16 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
 
             return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
         } catch (error) {
-            return await this._processAdditionalAction(
-                this._handleEmptyPaymentIntentError(error, stripeError),
-                methodId,
-                shouldSaveInstrument,
-                shouldSetAsDefaultInstrument
-            );
+            if (error instanceof Error) {
+                return await this._processAdditionalAction(
+                    this._handleEmptyPaymentIntentError(error, stripeError),
+                    methodId,
+                    shouldSaveInstrument,
+                    shouldSetAsDefaultInstrument
+                );
+            }
+
+            return Promise.resolve(this._store.getState());
```

Comment:
As you're already in an async block

```suggestion
            return this._store.getState();
```

Reply:
> Can you also double check that the output of `api-extractor` is good? I remember last time I had difficulty doing the upgrade.

Yes, that is something that I am verifying as of now.

Reply:
Have changed it to `throw error` instead.